### ISSUE TITLE
conda breaks when failing to install a package

### ISSUE THREAD

**Issue by SteveDoyle2** [conda/conda#2003]
```
C:\NASA\m4\formats\git\pyNastran\docs_sphinx>conda install mayavi
Fetching package metadata: ......
Solving package specifications: .....................
Package plan for installation in environment C:\Anaconda:

The following packages will be downloaded:

package                    |            build
---------------------------|-----------------
openssl-1.0.2e             |            vc9_1         4.4 MB
python-2.7.11              |                1        17.7 MB
numpy-1.9.3                |           py27_0        23.5 MB
setuptools-19.4            |           py27_0         670 KB
traits-4.4.0               |           py27_0         373 KB
pip-8.0.1                  |           py27_0         1.5 MB
pyface-4.4.0               |           py27_0         690 KB
traitsui-4.4.0             |           py27_0         833 KB
apptools-4.2.1             |           py27_0         282 KB
envisage-4.4.0             |           py27_1         267 KB
mayavi-4.4.0               |       np19py27_0        10.4 MB
------------------------------------------------------------
                                       Total:        60.6 MB

The following NEW packages will be INSTALLED:

apptools:   4.2.1-py27_0
envisage:   4.4.0-py27_1
mayavi:     4.4.0-np19py27_0
pyface:     4.4.0-py27_0
traits:     4.4.0-py27_0
traitsui:   4.4.0-py27_0

The following packages will be UPDATED:

openssl:    1.0.2d-vc9_0  [vc9] --> 1.0.2e-vc9_1     [vc9]
pip:        7.1.2-py27_0  --> 8.0.1-py27_0
python:     2.7.11-0      --> 2.7.11-1
setuptools: 19.1.1-py27_0 --> 19.4-py27_0

The following packages will be DOWNGRADED:

numpy:      1.10.1-py27_0 --> 1.9.3-py27_0

Proceed ([y]/n)? y

Fetching packages ...
openssl-1.0.2e 100% |###############################| Time: 0:00:01   3.54 MB/s
python-2.7.11- 100% |###############################| Time: 0:00:03   5.56 MB/s
numpy-1.9.3-py 100% |###############################| Time: 0:00:05   4.76 MB/s
setuptools-19. 100% |###############################| Time: 0:00:00   1.15 MB/s
traits-4.4.0-p 100% |###############################| Time: 0:00:00 697.60 kB/s
pip-8.0.1-py27 100% |###############################| Time: 0:00:00   1.85 MB/s
pyface-4.4.0-p 100% |###############################| Time: 0:00:00 910.01 kB/s
traitsui-4.4.0 100% |###############################| Time: 0:00:00   1.35 MB/s
apptools-4.2.1 100% |###############################| Time: 0:00:00 682.82 kB/s
envisage-4.4.0 100% |###############################| Time: 0:00:00 609.38 kB/s
mayavi-4.4.0-n 100% |###############################| Time: 0:00:02   4.73 MB/s
Extracting packages ...
[      COMPLETE      ]|##################################################| 100%
Unlinking packages ...
[      COMPLETE      ]|##################################################| 100%
Linking packages ...
An unexpected error has occurred, please consider sending the            |   9%
following traceback to the conda GitHub issue tracker at:

https://github.com/conda/conda/issues

Include the output of the command 'conda info' in your report.


Traceback (most recent call last):
  File "C:\Anaconda\Scripts\conda-script.py", line 4, in <module>
    sys.exit(main())
  File "C:\Anaconda\lib\site-packages\conda\cli\main.py", line 195, in main
    args_func(args, p)
  File "C:\Anaconda\lib\site-packages\conda\cli\main.py", line 202, in args_func
    args.func(args, p)
  File "C:\Anaconda\lib\site-packages\conda\cli\main_install.py", line 45, in execute
    install.install(args, parser, 'install')
  File "C:\Anaconda\lib\site-packages\conda\cli\install.py", line 423, in install
    plan.execute_actions(actions, index, verbose=not args.quiet)
  File "C:\Anaconda\lib\site-packages\conda\plan.py", line 537, in execute_actions
    inst.execute_instructions(plan, index, verbose)
  File "C:\Anaconda\lib\site-packages\conda\instructions.py", line 148, in execute_instructions
    cmd(state, arg)
  File "C:\Anaconda\lib\site-packages\conda\instructions.py", line 91, in LINK_CMD
    link(state['prefix'], arg, index=state['index'])
  File "C:\Anaconda\lib\site-packages\conda\instructions.py", line 87, in link
    install.link(pkgs_dir, prefix, dist, lt, index=index)
  File "C:\Anaconda\lib\site-packages\conda\install.py", line 616, in link
    os.makedirs(dst_dir)
  File "C:\Anaconda\lib\os.py", line 157, in makedirs
WindowsError: [Error 5] Access is denied: 'C:\\Anaconda\\DLLs'
```

Afterwards, no conda commands will work.

**Comment by SteveDoyle2** [comment]
Also, I think this was caused by Python being active in my IDE when I went to go update.

**Comment by kalefranz** [comment]
Fixed with transactions in conda 4.3.  No more conda uninstalling itself!!!
